### PR TITLE
feat(API,APP,CLI) changed `vector2`, `vector3`, `reserved`, `technical` and `description` formats

### LINKED ISSUE #363: Change attribute's vector format to array
Object attributes with type vector3 or vector2 expect a string formatted as a dictionary with coordinates (x,y,z) as keys and floats as values. The proposed change is to expect a string formatted as an array of floats where each element will respectively correspond to x, y and z. Spaces between numbers and commas are accepted and should be ignored. Examples:

- `"posXYZ" : "{"x":45 ,"y":50 ,"z":35}"` becomes `"posXYZ" : "[45, 50, 35]"`
- `"size" : "{"x":45 ,"y":50}"` becomes `"size" : "[45, 50]"`

This impacts CLI, API and 3D.

### LINKED ISSUE #366: Change description format from array to string
Currently an array of strings, description should be a simple string with no limit of chars and that accepts \n which will be replaced by a new line in the frontends.

### PR DESCRIPTION
**Pull request by Barbozafab**
## Description

- The `vector2` format was changed from `{"x":x, "y":y}` to `[x, y]`

- The `vector3` format was changed from `{"x":x, "y":y, "z":z}` to `[x, y, z]`

- The `reserved` and `technical` format was changed from `{"left":x, "right":y, "top":z, "bottom":w}` to `[z, w, y, x]`

- The `description` format was changed from an array of strings to a string with multiple lines separated by `\n`

- Fixes #363

- Fixes #366

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

- Local test

### CODE REVIEW

File: CLI/controllers/sendSchemaController.go [hunk 1]
```diff
@@ -20,15 +20,11 @@ func serialiseVector(attributes map[string]interface{}, attributeName string) {
...
-// the team wants it for now
-// "size":"[25,29.4,0]" -> "size": "{\"x\":25,\"y\":29.4,\"z\":0}"
 func serialiseStringVector(attr map[string]interface{}, want string) string {
 	var newSize string
 	if size, ok := attr[want].(string); ok {
 		left := strings.Index(size, "[")
 		right := strings.Index(size, "]")
-		coords := []string{"x", "y", "z"}
 
 		if left != -1 && right != -1 {
```
Comment (helderbetiol) [inline CLI/controllers/sendSchemaController.go:29]: Can't we simplify this now?

File: API/models/schemas/refs/types.json [hunk 2]
```diff
@@ -22,17 +22,17 @@
             "enum": ["mm", "cm", "m", "f", "U", "OU"]
         },
         "vector2": {
-            "pattern": "^\\s*{\\s*\"x\"\\s*:\\s*((-|\\+)?[0-9]*[.]?)?[0-9]+\\s*,\\s*\"y\"\\s*:\\s*((-|\\+)?[0-9]*[.]?)?[0-9]+\\s*}\\s*$",
```
[the remark this answers is not in the document]
Comment (Barbozafab) [reply API/models/schemas/refs/types.json:25]: I updated the patterns. They are now "standardised" and more readable.

File: CLI/controllers/sendSchemaController.go (hunk 1 above)
Comment (Barbozafab) [reply CLI/controllers/sendSchemaController.go]: I've rewritten the `serialiseStringVector` and `serialiseFloatVector` so they are a bit simpler and easier to understand now. Their functioning is still the same: extracting some attributes information (such as height, diameter etc.) from the vectors, and returning a vector (in form of a string) with only the necessary elements.

File: CLI/controllers/sendSchemaController.go [hunk 3]
```diff
@@ -20,106 +20,86 @@ func serialiseVector(attributes map[string]interface{}, attributeName string) {
...
 
-		if left != -1 && right != -1 {
-			var length int
-			subStr := size[left+1 : right]
-			nums := stringSplitter(subStr, ",", want)
-			if nums == nil { //Error!
-				return ""
-			}
-			//nums := strings.Split(subStr, ",")
+	size, ok := attr[want].(string)
```
Comment (helderbetiol) [inline CLI/controllers/sendSchemaController.go:26]: is it always size, though?

File: CLI/controllers/sendSchemaController.go [hunk 4]
```diff
@@ -20,106 +20,86 @@ func serialiseVector(attributes map[string]interface{}, attributeName string) {
...
+	}
+	//nums := strings.Split(subStr, ",")
 
-				if idx < length-1 {
-					newSize += ","
-				}
-			}
-			newSize = "{" + newSize + "}"
+	length := len(nums)
+	if length == 3 && want == "size" {
```
Comment (helderbetiol) [inline CLI/controllers/sendSchemaController.go:44]: stringSplitter won't allow this

File: CLI/controllers/sendSchemaController.go [hunk 5]
```diff
@@ -20,106 +20,86 @@ func serialiseVector(attributes map[string]interface{}, attributeName string) {
...
-			length = 2
-		} else {
-			//Make posXYZ length 3 for racks
-			if want == "posXYZ" && len(items) == 2 {
-				items = append(items, 0)
-			}
-			length = len(items)
-		}
+	length := len(items)
+	if length == 3 && want == "size" {
```
Comment (helderbetiol) [inline CLI/controllers/sendSchemaController.go:77]: Basically repeats the code from above, are we sure both are used? do we need both?

File: CLI/controllers/sendSchemaController.go (hunk 3 above)
Comment (Barbozafab) [reply CLI/controllers/sendSchemaController.go]: Not always... I've changed it to "vector".

File: CLI/controllers/sendSchemaController.go (hunk 4 above)
Comment (Barbozafab) [reply CLI/controllers/sendSchemaController.go]: Yup, that's true. Thank you! :)

File: CLI/controllers/sendSchemaController.go (hunk 5 above)
Comment (Barbozafab) [reply CLI/controllers/sendSchemaController.go]: Both are indeed used, but your review gave me an idea and I've managed to unify both functions thanks to the handy `Stringify` function in the same file. :)

**Pull request merged**